src/test/java/com/wikia/webdriver/testcases/commentstests/ArticleCommentsTests.java

```diff
@@ -4,34 +4,59 @@
 import com.wikia.webdriver.common.core.Assertion;
 import com.wikia.webdriver.common.core.annotations.Execute;
 import com.wikia.webdriver.common.core.api.ArticleContent;
-import com.wikia.webdriver.common.core.configuration.Configuration;
 import com.wikia.webdriver.common.core.helpers.User;
-import com.wikia.webdriver.common.properties.Credentials;
+import com.wikia.webdriver.common.logging.PageObjectLogging;
 import com.wikia.webdriver.common.templates.NewTestTemplate;
 import com.wikia.webdriver.pageobjectsfactory.componentobject.minieditor.MiniEditorComponentObject;
 import com.wikia.webdriver.pageobjectsfactory.pageobject.actions.DeletePageObject;
 import com.wikia.webdriver.pageobjectsfactory.pageobject.article.ArticlePageObject;
 
 import org.joda.time.DateTime;
+import org.openqa.selenium.TimeoutException;
+import org.testng.annotations.BeforeMethod;
 import org.testng.annotations.Test;
 
+import java.lang.reflect.Method;
+
 @Test(groups = "comments-articleComments")
 public class ArticleCommentsTests extends NewTestTemplate {
+  private ArticlePageObject article;
+
+  private String userName;
+
+  /**
+   * Setup test environment - navigate to article page
+   */
+  @BeforeMethod(alwaysRun = true)
+  public void initArticlePageAndCheckLoginStatus(Method method) {
+    User user = method.getAnnotation(Execute.class).asUser();
+
+    new ArticleContent().push(PageContent.ARTICLE_TEXT);
+    article = new ArticlePageObject().open();
+
+    if (user != User.ANONYMOUS) {
+      userName = user.getUserName();
 
-  Credentials credentials = Configuration.getCredentials();
+      // SUS-1737: In Comments tests sometimes user is not authenticated properly
+      // Let's check for this, and if this happens, log the situation and log in again
+      try {
+        article.verifyUserLoggedIn(user);
+      } catch (TimeoutException e) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-s4/coding_rules#rule_key=squid%3AS1166)